Sources/KlaviyoSwift/Klaviyo.swift

```diff
@@ -466,14 +475,18 @@ public struct KlaviyoSDK {
     ///   - remoteNotification: the remote notificaiton that was opened
     ///   - fetchCompletionHandler: a completion handler that will be called with a result for Klaviyo notifications
     /// - Returns: true if the notificaiton originated from Klaviyo, false otherwise.
-    public func handle(notificationResponse: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) -> Bool {
+    public func handle(notificationResponse: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void, deepLinkHandler: ((URL) -> Void)? = nil) -> Bool {
         if let properties = notificationResponse.notification.request.content.userInfo as? [String: Any],
            let body = properties["body"] as? [String: Any], let _ = body["_k"] {
             create(event: Event(name: .OpenedPush, properties: properties, profile: [:]))
             Task {
                 await MainActor.run {
                     if let url = properties["url"] as? String, let url = URL(string: url) {
-                        UIApplication.shared.open(url)
+                        if let deepLinkHandler = deepLinkHandler {
+                            deepLinkHandler(url)
+                        } else {
+                            UIApplication.shared.open(url)
+                        }
                     }
                     completionHandler()
```

Comment:
Maybe just a nit, but does it make a difference that you're running the completion handler on main thread here vs how you left it to the dev to call completionHandler before?

Reply:
This part was always on the main thread. If you are talking about the code above though I think it should have been on the main thread before. I suspect that function is always called on the main thread anyway but putting it back on it is just a safety measure since devs are likely to call ui related code in the callback.